tests/run/tupled-function-andThen.scala

```diff
@@ -0,0 +1,27 @@
+object Test {
+  def main(args: Array[String]): Unit = {
+
+    val f1 = (x1: Int, x2: Int) => (x1, x2, x1 + x2)
+    val g1 = (x1: Int, x2: Int, x3: Int) => x1 + x2 + x3
+    val h1 = f1.tupled.andThen(g1.tupled)
```

Comment:
It’s not clear from this PR that `.tupled` is related to `TupledFunction`. It seems to just return a `Function1`?

Reply:
Yes, those are on `Function1`. Originally I had tests for a generalized `andThen` and `compose` (removed in https://github.com/lampepfl/dotty/commit/92c83c14634d022093f7d0740009a42f9b823baa#diff-a92ddcc294d7060f5b3e9e8c84b03997L22). I should probably add them back.

Reply:
I re-enabled the original `andThen` and `compose` extension. Now the test looks like 
```scala
    val f1 = (x1: Int, x2: Int) => (x1, x2, x1 + x2)
    val g1 = (x1: Int, x2: Int, x3: Int) => x1 + x2 + x3
    val h1 = f1.andThen(g1)
```